strings/base_implements.h

```diff
@@ -1033,8 +1039,10 @@ namespace winrt::impl
                     {
                         return error_bad_alloc;
                     }
-                    *array = std::copy(local_iids.second, local_iids.second + local_count, *array);
-                    std::copy(inner_iids.cbegin(), inner_iids.cend(), *array);
+                    auto _array = *array;
+                    _array = std::copy(local_iids.second, local_iids.second + local_iids.first, _array);
+                    _array = std::copy(tearoff_iids.cbegin(), tearoff_iids.cend(), _array);
```

Comment:
Do these copies go to the unoccupied slots in the array or do they keep overwriting the first elements?  As best as I can tell `_array` is a pointer, not an iterator, so copy wouldn't be advancing it.

Reply:
`std::copy` returns a pointer to the next empty element. So `_array` ensures we don't write over the beginning of `*array` with each successive copy.

Reply:
Ah, I missed where the _array pointer is updated to be the output of `std::copy`.